reaction_network/fjss.py

```diff
@@ -738,74 +738,43 @@ def build_model_gurobi(self):
 
                 for j in range(n_ws):
                     model.addConstr(
-                        var_c[i] - var_s[i] <= self.workshifts[j][0], name="workshift_duration_limit_{i}"
+                        var_c[i] - var_s[i] <= self.workshifts[j][0], name="workshift_duration_limit_{i, j}"
                         )
 
-                    # s_i <= (i+1)*shift_duration and c_i <= (i+1)*shift_duration
-                    var_auxiliary_a = model.addVar(
-                        vtype=GRB.BINARY, name=f"var_auxiliary_a_{i,j}"
-                    )
-                    var_auxiliary_b = model.addVar(
-                        vtype=GRB.BINARY, name=f"var_auxiliary_b_{i,j}"
-                    )
-
-                    # var_auxiliary_a indicates j-th shift starting time >= s_i
-                    model.addConstr(
-                        self.ws_starting_time[j] >= var_s[i] + eps - value_m * ( 1- var_auxiliary_a)
-                    )
+                    # the i-th operation can be only be assigned to one work shift
                     model.addConstr(
-                        self.ws_starting_time[j] <= var_s[i] + value_m * var_auxiliary_a
-                    )
+                        gp.quicksum(var_ws_assignments[i, j] for j in range(n_ws)) == 1, name="workshift_assignment_{i,j}"
```

Comment:
j reused here (you used it in L739)

Reply:
```math
\sum_j v_{ij} = 1, \forall i 
```
this constraint should certainly be outside the loop of L739

Reply:
Fixed.